igor-web/src/main/groovy/com/netflix/spinnaker/igor/gitlabci/service/GitlabCiService.java

```diff
@@ -94,37 +119,116 @@ public Permissions getPermissions() {
+  }
+
+  // Gets a pipeline's jobs along with any child pipeline jobs (bridges)
+  private List<Job> getJobsWithBridges(String projectId, Integer pipelineId) {
+    List<Job> jobs = this.client.getJobs(projectId, pipelineId);
+    List<Bridge> bridges = this.client.getBridges(projectId, pipelineId);
+    bridges.parallelStream()
+        .filter(
+            bridge -> {
+              // Filter out any child pipelines that failed or are still in-progress
+              Pipeline parent = bridge.getDownstreamPipeline();
+              return parent != null
+                  && GitlabCiResultConverter.getResultFromGitlabCiState(parent.getStatus())
+                      == Result.SUCCESS;
+            })
+        .forEach(
+            bridge -> {
+              jobs.addAll(this.client.getJobs(projectId, bridge.getDownstreamPipeline().getId()));
+            });
+    return jobs;
+  }
+
+  private Map<String, Object> getPropertyFileFromLog(String projectId, Integer pipelineId) {
+    Map<String, Object> properties = new HashMap<>();
+    return retrySupport.retry(
+        () -> {
+          try {
+            Pipeline pipeline = this.client.getPipeline(projectId, pipelineId);
+            PipelineStatus status = pipeline.getStatus();
+            if (status != PipelineStatus.running) {
+              log.error(
+                  "Unable to get GitLab build properties, pipeline '{}' in project '{}' has status {}",
+                  kv("pipeline", pipelineId),
+                  kv("project", projectId),
+                  kv("status", status));
+            }
+            // Pipeline has many jobs, jobs have many artifacts
+            // No way to list all job's artifacts so loop the jobs
+            // trying to find the artifact.  Return if/when one is found
+            List<Job> jobs = getJobsWithBridges(projectId, pipelineId);
+            for (Job job : jobs) {
+              InputStream logStream = this.client.getJobLog(projectId, job.getId()).getBody().in();
+              String log = new String(logStream.readAllBytes(), StandardCharsets.UTF_8);
+              Map<String, Object> jobProperties = PropertyParser.extractPropertiesFromLog(log);
+              properties.putAll(jobProperties);
+            }
+
+            return properties;
+
+          } catch (RetrofitError e) {
+            // retry on network issue, 404 and 5XX
+            if (e.getKind() == RetrofitError.Kind.NETWORK
+                || (e.getKind() == RetrofitError.Kind.HTTP
+                    && (e.getResponse().getStatus() == 404
+                        || e.getResponse().getStatus() >= 500))) {
+              throw e;
+            }
+            SpinnakerException ex = new SpinnakerException(e);
+            ex.setRetryable(false);
+            throw ex;
+          } catch (IOException e) {
+            log.error("Error while parsing GitLab CI log to build properties", e);
+            return properties;
+          }
+        },
+        5,
```

Comment:
Make these configurable?  Or at least add comments explaining what the params are, and how you chose these values?

Reply:
made configurable (original value was copied from another polling monitor)

Reply:
Haven't seen a commit arrive with these changes...

Reply:
Whoops...I see it now.